python/lsst/pipe/tasks/dcrAssembleCoadd.py

```diff
@@ -561,49 +555,43 @@ def dcrAssembleSubregion(self, dcrModels, bbox, dcrBBox, tempExpRefList, imageSc
         modelWeights : `numpy.ndarray` or `float`
             A 2D array of weight values that tapers smoothly to zero away from detected sources.
             Set to a placeholder value of 1.0 if ``self.config.useModelWeights`` is False.
-        refImage : `lsst.afw.image.MaskedImage`
+        refImage : `lsst.afw.image.Image`
             A reference image used to supply the default pixel values.
+        dcrWeights : `list` of `lsst.afw.image.Image`
+            Per-pixel weights for each subfilter.
+            Equal to the number of unmasked images contributing to each pixel.
         """
-        tempExpName = self.getTempExpDatasetName(self.warpType)
         residualGeneratorList = []
 
-        for tempExpRef, imageScaler, altMaskSpans in zip(tempExpRefList, imageScalerList, spanSetMaskList):
-            exposure = tempExpRef.get(tempExpName + "_sub", bbox=dcrBBox)
+        for tempExpRef in tempExpRefList:
+            exposure = subExposures[tempExpRef.dataId["visit"]]
             visitInfo = exposure.getInfo().getVisitInfo()
             wcs = exposure.getInfo().getWcs()
-            maskedImage = exposure.maskedImage
-            templateImage = dcrModels.buildMatchedTemplate(warpCtrl=self.warpCtrl, visitInfo=visitInfo,
-                                                           bbox=dcrBBox, wcs=wcs, mask=baseMask,
+            templateImage = dcrModels.buildMatchedTemplate(exposure=exposure,
+                                                           order=self.config.imageInterpOrder,
                                                            splitSubfilters=self.config.splitSubfilters)
-            imageScaler.scaleMaskedImage(maskedImage)
-            if altMaskSpans is not None:
-                self.applyAltMaskPlanes(maskedImage.mask, altMaskSpans)
+            residual = exposure.image.array - templateImage.array
+            # Note that the variance plane here is used to store weights, not the actual variance
+            residual *= exposure.variance.array
+            residualGeneratorList.append(self.dcrResiduals(residual, visitInfo, wcs, dcrModels.filter))
```

Comment:
`residualGeneratorList` sounds like a `generator` that produces `list`s or something?

Reply:
It's a list of generators, and I agree that it is confusing but this seemed to be the most memory-efficient way to pass nSubfilter X nImage residual arrays. I can add an additional comment to explain it.